frontend/src/app/fullname.pipe.ts

```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { PersonModel } from './models/person.model';
+
+@Pipe({
+  name: 'fullname'
+})
+export class FullnamePipe implements PipeTransform {
+
+  transform(person: PersonModel): string {
+    const elements = [person.firstName, person.lastName, person.nickName ? `(${person.nickName})` : null];
+    return elements.filter(e => e).join(' ');
```

Comment:
Funny to do it this way :) Took me a few seconds to understand what was going on. 

I would have simply returned: 

```typescript
const nickName = person.nickName ? ` (${person.nickName})` : '';
return `${person.firstName} ${person.lastName}${nickname}`;
```

Less pretty maybe.